tests/health_mgmtapi/health-cmdapi-test.sh.in

```diff
@@ -1,58 +1,78 @@
+	-t TEST scenarios execution
+	-u <URL> changes the host:port from localhost:19999 to <URL>
+	"
+}
+
 check () {
-	sec=1
-	echo -e "  ${GRAY}Check: '${1}' in $sec sec"
-	sleep $sec
-    number=$RANDOM
-	resp=$(curl -s "http://$URL/api/v1/alarms?all&$number")
-	r=$(echo "${resp}" | \
-	python3 -c "import sys, json; d=json.load(sys.stdin); \
-	print(\
-	d['alarms']['system.cpu.10min_cpu_usage']['disabled'], \
-	d['alarms']['system.cpu.10min_cpu_usage']['silenced'] , \
-	d['alarms']['system.cpu.10min_cpu_iowait']['disabled'], \
-	d['alarms']['system.cpu.10min_cpu_iowait']['silenced'], \
-	d['alarms']['system.load.load_trigger']['disabled'], \
-	d['alarms']['system.load.load_trigger']['silenced'], \
-	);" 2>&1)
-	if [ $? -ne 0 ] ; then
-		echo -e "  ${RED}ERROR: Unexpected response stored in /tmp/resp-$number.json"
-		echo "$resp" > /tmp/resp-$number.json
-		err=$((err+1))
-		iter=0
-	elif [ "${r}" != "${2}" ] ; then
-		echo -e "  ${GRAY}WARNING: 'Got ${r}'. Expected '${2}'"
-		iter=$((iter+1))
-		if [ $iter -lt 10 ] ; then 
-			echo -e "  ${GRAY}Repeating test "
-			check "$1" "$2"
+	iterations=1
+	error=1
+	while [ ${iterations} -lt 6 ] && [ ${error} -eq 1 ]; do
+		echo -e "${GRAY}Check (${iterations}/5): '${1}' in 1 sec"
+		sleep 1
+		resp=$(curl -s "http://$URL/api/v1/alarms?all")
+		r=$(echo "${resp}" | \
+		python3 -c "import sys, json; d=json.load(sys.stdin); \
+		print(\
+		d['alarms']['example.random.example_alarm1']['disabled'], \
+		d['alarms']['example.random.example_alarm1']['silenced'] , \
+		d['alarms']['example.random.example_alarm2']['disabled'], \
+		d['alarms']['example.random.example_alarm2']['silenced'], \
+		d['alarms']['system.load.load_trigger']['disabled'], \
+		d['alarms']['system.load.load_trigger']['silenced'], \
+		);" 2>&1)
+		if [ $? -ne 0 ] ; then
+			echo -e "${RED}  ERROR: Unexpected response"
+		elif [ "${r}" != "${2}" ] ; then
+			echo -e "${RED}  ERROR: 'Got ${r}'. Expected '${2}'"
 		else
-			echo -e "  ${RED}ERROR: 'Got ${r}'. Expected '${2}'"
-			iter=0
-			err=$((err+1))
+			echo -e "${GREEN}Success"
+			error=0
 		fi
-	else
-		echo -e "  ${GREEN}Success"
-		iter=0
+		iterations=$((iterations+1))
+	done
+	if [ ${error} -eq 1 ] ; then
+		err=$((err+1));
+		echo -e "${RED}Failure"
 	fi
 }
 
 cmd () {
-	echo -e "${WHITE}Cmd '${1}'"
-	echo -en "  ${GRAY}Expecting '${2}' : "
+	echo -e "${WHITE}Cmd '${1}', expecting '${2}'"
 	RESPONSE=$(curl -s "http://$URL/api/v1/manage/health?${1}" -H "X-Auth-Token: $TOKEN" 2>&1)
 	if [ "${RESPONSE}" != "${2}" ] ; then
-		echo -e "${RED}ERROR: Response '${RESPONSE}'"
+		echo -e "${RED}ERROR: Response '${RESPONSE}' != '${2}'"
 		err=$((err+1))
 	else
 		echo -e "${GREEN}Success"
 	fi
 }
 
+check_list() {
+    RESPONSE=$(curl -s "http://$URL/api/v1/manage/health?cmd=LIST" -H "X-Auth-Token: $TOKEN" 2>&1)
+    SERVER=$( echo "${RESPONSE}" | python3 -c "import sys,json; import netdata_json as nj; e=json.load(sys.stdin); d=nj.orderme(e); print(d)" 2>&1)
```

Comment:
```
++ python3 -c 'import sys,json; import netdata_json as nj; e=json.load(sys.stdin); d=nj.orderme(e); print(d)'
+ SERVER='Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named '\''netdata_json'\'''
```

And then the script got stuck forever. Based on the other comment I made, you'll get rid of this parsing completely, since you'll only be doing `diff file1 file2` and checking the output.

Reply:
I used python to order the json and to be sure that what is written in the disk is the same response of the server, this is the motive I decided to give an extra work with the input and output.